File: BE/src/main/java/com/example/be/service/AuthService.java
package com.example.be.service;

import com.example.be.dto.auth.JwtResponse;
import com.example.be.dto.auth.LoginRequest;
import com.example.be.dto.auth.RegisterRequest;

public interface AuthService {
    JwtResponse login(LoginRequest loginRequest);

    String register(RegisterRequest registerRequest);
}

File: FE/src/services/authService.ts
import api from './api'
import type { LoginRequest, RegisterRequest, JwtResponse } from '@/types/auth' // We'll need to define these types

export interface LoginPayload {
  username: string
  password: string
}

export interface RegisterPayload {
  username: string
  password: string
  email: string
  fullName: string
}

export interface AuthResponse {
  accessToken: string
  tokenType: string
  user: any // UserDTO
}

class AuthService {
  async login(payload: LoginPayload): Promise<AuthResponse> {
    const response = await api.post('/auth/login', payload)
    return response.data
  }

  async register(payload: RegisterPayload): Promise<string> {
    const response = await api.post('/auth/register', payload)
    return response.data
  }

  logout() {
    localStorage.removeItem('token')
    localStorage.removeItem('user')
  }
}

export default new AuthService()
